src/rendering/vcTileRenderer.cpp

```diff
@@ -304,53 +308,88 @@ uint32_t vcTileRenderer_LoadThread(void *pThreadData)
       {
-        udSprintf(localURL, "%s/%s/%d/%d/%d.png", pRenderer->pSettings->cacheAssetPath, udUUID_GetAsString(demTileServerAddresUUID), pBestNode->slippyPosition.z, pBestNode->slippyPosition.x, pBestNode->slippyPosition.y);
-        udSprintf(serverURL, pDemTileServerAddress, pBestNode->slippyPosition.z, pBestNode->slippyPosition.x, pBestNode->slippyPosition.y);
-
-        // allow continue on failure
-        demResult = vcTileRenderer_HandleTileDownload(&pBestNode->demInfo, serverURL, localURL);
+        udSprintf(localURLDEM, "%s/%s/%d/%d/%d.png", pRenderer->pSettings->cacheAssetPath, udUUID_GetAsString(demTileServerAddresUUID), pBestNode->slippyPosition.z, pBestNode->slippyPosition.x, pBestNode->slippyPosition.y);
+        udSprintf(serverURLDEM, pDemTileServerAddress, pBestNode->slippyPosition.z, pBestNode->slippyPosition.x, pBestNode->slippyPosition.y);
       }
 
-      if (pBestNode->colourInfo.loadStatus.Get() == vcNodeRenderInfo::vcTLS_Downloading)
+      if (canDownloadColour)
       {
-        udSprintf(localURL, "%s/%s/%d/%d/%d.png", pRenderer->pSettings->cacheAssetPath, udUUID_GetAsString(pRenderer->pSettings->maptiles.activeServer.tileServerAddressUUID), pBestNode->slippyPosition.z, pBestNode->slippyPosition.x, pBestNode->slippyPosition.y);
+        udSprintf(localURLColour, "%s/%s/%d/%d/%d.png", pRenderer->pSettings->cacheAssetPath, udUUID_GetAsString(pRenderer->pSettings->maptiles.activeServer.tileServerAddressUUID), pBestNode->slippyPosition.z, pBestNode->slippyPosition.x, pBestNode->slippyPosition.y);
 
         udStrItoa(xSlippyStr, pBestNode->slippyPosition.x);
         udStrItoa(ySlippyStr, pBestNode->slippyPosition.y);
         udStrItoa(zSlippyStr, pBestNode->slippyPosition.z);
-        
-        vcStringFormat(serverURL, udLengthOf(serverURL), pRenderer->pSettings->maptiles.activeServer.tileServerAddress, pSlippyStrs, udLengthOf(pSlippyStrs));
 
-        // allow continue on failure
-        colourResult = vcTileRenderer_HandleTileDownload(&pBestNode->colourInfo, serverURL, localURL);
+        vcStringFormat(serverURLColour, udLengthOf(serverURLColour), pRenderer->pSettings->maptiles.activeServer.tileServerAddress, pSlippyStrs, udLengthOf(pSlippyStrs));
       }
 
-      if (demResult != udR_Success)
+      //We release the mutex to allow work to continue on the quadtree.
+      //For instance, the quadtree may be cleared duting the loads below.
+      udReleaseMutex(pCache->pMutex);
+
+      ProxyTexture textureDEM = {};
+      ProxyTexture textureColour = {};
+
+      if (canDownloadDEM)
+        demResult = vcTileRenderer_HandleTileDownload(&textureDEM, serverURLDEM, localURLDEM);
+
+      if (canDownloadColour)
+        colourResult = vcTileRenderer_HandleTileDownload(&textureColour, serverURLColour, localURLColour);
+
+      udLockMutex(pCache->pMutex);
+
+      //Check if the node is still valid.
+      if (canDownloadDEM && pBestNode->demInfo.loadStatus.Get() == vcNodeRenderInfo::vcTLS_Downloading)
```

Comment:
The thread should assign ownership to `pBestNode`. Otherwise while this thread is loading data, the node could be cleared and taken by another thread and subsequently flagged as `vcTLS_Downloading`. Now we have two threads trying to populate in this node.